parsing/parser.mly

```diff
@@ -2309,6 +2309,24 @@ simple_expr:
   | LPAREN MODULE ext_attributes module_expr COLON error
       { unclosed "(" $loc($1) ")" $loc($6) }
 ;
+
+inner_comprehension:
+|  ext_attributes pattern EQUAL simple_expr direction_flag simple_expr
+      { From_to($2, $4, $6, $5) }
+|  ext_attributes pattern IN simple_expr { In($2, $4) }
+;
+
+// List are reveresed s.t. we can Left fold over it in the typecore.
```

Comment:
Since these lists will not be long, it's probably better to store them in the AST in the order that someone would expect rather than reversed.